### PR TITLE
Alex/verify lead

### PR DESCRIPTION
**Pull request by FluffyPenguin**
# Description

Incoming leads to api/webhook are now verified by google key and made sure they already are not received
- On bad google key - returns 401 unauthorized
- On duplicate lead (lead already exists) - returns 409 Conflict

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Manually in local development server

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java [hunk 1]
```diff
@@ -116,7 +119,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     Lead newLead = Lead.fromReader(request.getReader(), advertiserKey);
 
-    //verify the lead
```
Comment (ataddesse) [inline backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java:119]: Enhance your documentation here.

File: backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java [hunk 2]
```diff
@@ -116,7 +119,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      response.sendError(401, "The provided Google Key and webhook do not match.");
+      return;
+    }
+
+    //de-duplicate the lead
+    Key leadKey = Lead.generateKey(advertiserKey, newLead.getLeadId());
+    if (DatastoreUtil.exists(leadKey)) {
+      response.sendError(409, "This lead already has been successfully sent and exists.");
+      return;
+    }
```
Comment (ataddesse) [inline backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java:134]: Delete one empty line here.

File: backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java (hunk 1 above)
Comment (FluffyPenguin) [reply backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java:119]: see below:     //verify the lead - has correct google key for this advertiser. does it need more explanation?

File: backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java (hunk 2 above)
Comment (FluffyPenguin) [reply backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java:134]: fixed

File: backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java (hunk 1 above)
Comment (ataddesse) [reply backend/src/main/java/com/google/minicrm/servlets/WebhookServlet.java:119]: Actually yea I see that you already explained well on the comment before the dopost method so this should be enough.

**Pull request merged**